eth_account/hdaccount/mnemonic.py

```diff
@@ -0,0 +1,161 @@
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+# of the Software, and to permit persons to whom the Software is furnished to do
+# so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import binascii
+import bisect
+import hashlib
+import os
+import unicodedata
+
+PBKDF2_ROUNDS = 2048
+
+
+class ConfigurationError(Exception):
+    pass
+
+
+# https://stackoverflow.com/questions/212358/binary-search-bisection-in-python/2233940#2233940
+def binary_search(a, x, lo=0, hi=None):  # can't use a to specify default for hi
+    hi = hi if hi is not None else len(a)  # hi defaults to len(a)
+    pos = bisect.bisect_left(a, x, lo, hi)  # find insertion position
+    return pos if pos != hi and a[pos] == x else -1  # don't walk off the end
+
+
+class Mnemonic(object):
+    def __init__(self, language):
+        self.radix = 2048
+        with open(f"{self._get_directory()}/{language}.txt", "r", encoding="utf-8") as f:
+            self.wordlist = [w.strip() for w in f.readlines()]
+        if len(self.wordlist) != self.radix:
+            raise ConfigurationError(
+                f"Wordlist should contain {self.radix} words, "
+                f"but it contains {len(self.wordlist)} words."
+            )
+
+    @classmethod
+    def _get_directory(cls):
+        return os.path.join(os.path.dirname(__file__), "wordlist")
+
+    @classmethod
+    def list_languages(cls):
+        return [
+            f.split(".")[0]
+            for f in os.listdir(cls._get_directory())
+            if f.endswith(".txt")
+        ]
+
+    @classmethod
+    def normalize_string(cls, txt):
+        if isinstance(txt, bytes):
+            utxt = txt.decode("utf8")
+        elif isinstance(txt, str):
+            utxt = txt
+        else:
+            raise TypeError("String value expected")
+
+        return unicodedata.normalize("NFKD", utxt)
+
+    @classmethod
+    def detect_language(cls, code):
+        code = cls.normalize_string(code)
+        first = code.split(" ")[0]
+        languages = cls.list_languages()
+
+        for lang in languages:
+            mnemo = cls(lang)
+            if first in mnemo.wordlist:
+                return lang
+
+        raise ConfigurationError("Language not detected")
+
+    def generate(self, strength=128):
```

Comment:
For #87, I used similar RNG setup as `Account.generate()` instead of what was used here. Was unsure if that was preferred or if we wants to give users more control over the amount of seed words they generate through this part of the library. Perhaps it is better to do the latter, I can refactor to do this by changing the RNG process to look more like what already existed.

Reply:
Part of this will be fixed in how API is used in #87